front/app/components/IdeasMap/index.tsx

```diff
@@ -1,160 +1,73 @@
 import React, {
   memo,
-  useState,
-  useRef,
+  useCallback,
```

Comment:
Might be easier to review this file just by looking at the final file in VS code or on GitHub instead :)